lib/libc/minimal/include/signal.h

```diff
@@ -0,0 +1,25 @@
+/*
+ * Copyright (c) 2023 Meta
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef ZEPHYR_LIB_LIBC_MINIMAL_INCLUDE_SIGNAL_H_
+#define ZEPHYR_LIB_LIBC_MINIMAL_INCLUDE_SIGNAL_H_
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+typedef int	sig_atomic_t;		/* Atomic entity type (ANSI) */
+
+/* Note: sighandler_t is a gnu-ism, but it simplifies the declaration below */
+typedef void (*sighandler_t)(int signo);
+
+sighandler_t signal(int signum, sighandler_t handler);
```

Comment:
Does minimal C library support this function? If not, you probably shouldn't declare it here.

Reply:
The C standard specifies behavior for the `raise()` and `abort()` functions related to `signal()`, so there are bits which "should" be provided as part of a C environment. (whether any of that is useful is another question). However, if you want to just get rid of the whole thing and pretend signals don't exist, then I'd suggest you might be better served by ignoring calls to signal rather than reporting an error. In this case, you're promising to call the specified function whenever the related signal is raised. As said signal will *never* be raised (unless the application uses `raise()` itself, then returning SIG_DFL would prevent applications that check the return value from failing while still meeting the API promise.

Reply:
Fixed! Should pass CI now, since unrelated errors are fixed now.